### PR TITLE
Bell/execution device

### PR DESCRIPTION
**Pull request by songbell**
### Details:
 - add property execution_device for auto users to query runtime target devices

### Tickets:
 - 92659

### CODE REVIEW

File: samples/cpp/benchmark_app/main.cpp [hunk 1]
```diff
@@ -1261,8 +1261,10 @@ int main(int argc, char* argv[]) {
             statistics->dump();
 
         // Performance metrics report
-        slog::info << "Count:      " << iteration << " iterations" << slog::endl;
-        slog::info << "Duration:   " << double_to_string(totalDuration) << " ms" << slog::endl;
+        if (device_name.find("AUTO") != std::string::npos)
```
Comment (yangwang201911) [inline samples/cpp/benchmark_app/main.cpp:1264]: Should we need to add warning that device switching will happen soon when retrieving the property ov::execution_devices at step 8?

**Review by yangwang201911 — APPROVED** [review]

File: src/plugins/auto/auto_schedule.cpp [hunk 2]
```diff
@@ -175,15 +174,18 @@ void AutoSchedule::init(const ScheduleContext::Ptr& sContext) {
     }
     LOG_INFO_TAG("select device:%s", _loadContext[ACTUALDEVICE].deviceInfo.deviceName.c_str());
     bool isActualDevCPU =
-        _loadContext[ACTUALDEVICE].deviceInfo.deviceName.find("CPU") !=std::string::npos;
+        _loadContext[ACTUALDEVICE].deviceInfo.deviceName.find("CPU") !=std::string::npos && !isCumulative;
     // if Actual device is CPU, disabled _loadContext[CPU], only use _loadContext[ACTUALDEVICE]
     if (isActualDevCPU || isCumulative) {
         _loadContext[CPU].isEnabled = false;
+        _autoSContext->_exeDevices = isActualDevCPU ? "CPU"
+                        : _loadContext[ACTUALDEVICE].deviceInfo.deviceName.substr(_loadContext[ACTUALDEVICE].deviceInfo.deviceName.find(":") + 1);
```
Comment (wangleis) [inline src/plugins/auto/auto_schedule.cpp:182]: In cumulative throughput mode, the executing device should remove the device that failed to load the network.

Comment (songbell) [reply src/plugins/auto/auto_schedule.cpp:182]: ctput will load to multi:*,* for multi, there is no logic to check load failure

Comment (wangleis) [reply src/plugins/auto/auto_schedule.cpp:182]: Will follow up in next PR

**Review by wangleis — APPROVED** [review]

File: samples/cpp/benchmark_app/main.cpp (hunk 1 above)
Comment (songbell) [reply samples/cpp/benchmark_app/main.cpp:1264]: the logic behind is quite a threading thing, the exe devices may change or not change depending on the load status in background threads

File: src/inference/include/ie/ie_plugin_config.hpp [hunk 3]
```diff
@@ -463,6 +463,13 @@ DECLARE_CONFIG_KEY(CACHE_DIR);
  */
 DECLARE_CONFIG_KEY(FORCE_TBB_TERMINATE);
 
+/**
+ * @brief The devices that the inference tasks been executed on.
+ *
+ * value type: string
+ *
+ */
+DECLARE_CONFIG_KEY(EXECUTION_DEVICES);
```
Comment (nkogteva) [inline src/inference/include/ie/ie_plugin_config.hpp:472]: Do we actually need this declaration? I think only new property would be introduced.

File: samples/cpp/benchmark_app/main.cpp [hunk 4]
```diff
@@ -1261,8 +1261,10 @@ int main(int argc, char* argv[]) {
...
+            slog::info << "ExecutionDevice: " << compiledModel.get_property(ov::execution_devices) << slog::endl;
+        slog::info << "Count:           " << iteration << " iterations" << slog::endl;
+        slog::info << "Duration:        " << double_to_string(totalDuration) << " ms" << slog::endl;
```
Comment (nkogteva) [inline samples/cpp/benchmark_app/main.cpp:1267]: What about python tool?

File: src/inference/include/openvino/runtime/properties.hpp [hunk 5]
```diff
@@ -884,4 +884,10 @@ inline std::istream& operator>>(std::istream& is, Affinity& affinity) {
  * environment variable is set (as affinity is configured explicitly)
  */
 static constexpr Property<Affinity> affinity{"AFFINITY"};
+
+/**
+ * @brief The devices that the inference task been executed.
+ * @ingroup ov_runtime_cpp_prop_api
+ */
+static constexpr Property<std::string> execution_devices{"EXECUTION_DEVICES"};
```
Comment (ilya-lavrenov) [inline src/inference/include/openvino/runtime/properties.hpp:892]: can we also support this for HETERO?

Comment (songbell) [reply src/inference/include/openvino/runtime/properties.hpp:892]: Hi ilya, for hetero, I want to test more with situations like 1) on multiple GPUs target 2) more hetero specific cases featured with affinity, I would prefer to track in a sperate PR with jira https://jira.devtools.intel.com/browse/CVS-92661

File: src/inference/include/ie/ie_plugin_config.hpp (hunk 3 above)
Comment (songbell) [reply src/inference/include/ie/ie_plugin_config.hpp:472]: fixed

File: samples/cpp/benchmark_app/main.cpp (hunk 4 above)
Comment (songbell) [reply samples/cpp/benchmark_app/main.cpp:1267]: updated python

**Review by nkogteva — APPROVED** [review]

**Pull request merged**